### PR TITLE
Language Server - Command class

### PR DESCRIPTION
**Pull request by fredg1**
Adds the Command class as an abstract subtype of ParseTreeNode.

Its existence was already assumed (shown with the presence of variable names and methods like "parseCommand", to spell it out loud), but never officially added.

This class creates a distinction between it and Symbol, and shields previous uses of "ParseTreeNode" from having a Symbol slipped in its place.

### CODE REVIEW

File: src/net/sourceforge/kolmafia/textui/Parser.java [hunk 1]
```diff
@@ -813,7 +813,7 @@ private Function parseFunction( final Type functionType, final Scope parentScope
 		Scope scope = this.parseBlockOrSingleCommand( functionType, paramList, parentScope, false, false, false );
 
 		result.setScope( scope );
-		if ( !result.assertBarrier() && !functionType.equals( DataTypes.TYPE_VOID ) )
+		if ( !scope.assertBarrier() && !functionType.equals( DataTypes.TYPE_VOID ) )
```
Comment (heeheehee-kolmafia) [inline src/net/sourceforge/kolmafia/textui/Parser.java:816]: What's the rationale for this change? Are we getting rid of UserDefinedFunction.assertBarrier()?

(once I got to the bottom of this review, signs point to "yes". I guess that's okay.)

File: src/net/sourceforge/kolmafia/textui/parsetree/Command.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package net.sourceforge.kolmafia.textui.parsetree;
+
+public abstract class Command
+	extends ParseTreeNode
+{
+	// A barrier is any code construct that is fundamentally incapable of
```
Comment (heeheehee-kolmafia) [inline src/net/sourceforge/kolmafia/textui/parsetree/Command.java:6]: Convert this to a multiline comment, and add blank lines between paragraphs? Otherwise this is just a wall of text that I don't feel like reading.

(ditto below.)

(Although, reading on, I see this was lifted directly from ParseTreeNode...)

**Review by heeheehee-kolmafia — APPROVED** [review]
I had written a couple of comments as I was reading, then went back and edited them for context (in case someone reading this in the future has the same questions I initially did).

One optional suggestion (reformatting some very long comments that were taken verbatim from ParseTreeNode), but otherwise looks good.

**Comment by jaadams5** [comment]
@heeheehee-kolmafia - do we not automatically have Squash and merge after an authorized review?

**Comment by heeheehee-kolmafia** [comment]
We do, but I'm leaving it open to @fredg1 to decide whether he wants to make the optional changes that I suggested, or merge them himself.

**Comment by fredg1** [comment]
> 
> 
> We do, but I'm leaving it open to @fredg1 to decide whether he wants to make the optional changes that I suggested, or merge them himself.

"Merging is blocked
The base branch restricts merging to authorized users. Learn more about protected branches. "

**Comment by heeheehee-kolmafia** [comment]
Ack. I'll merge it, then.

**Comment by jaadams5** [comment]
The question was about automation - at one point there was talk about a workflow that automatically squashed and merged once an authorized user approved in a review.  Since that had not happened I wondered why.  Does an approved reviewer also have to push a button?
